plugins/net_plugin/net_plugin.cpp

```diff
@@ -3983,6 +4015,29 @@ namespace eosio {
       return fc::json::from_string(s).as<T>();
    }
 
+   size_t net_plugin_impl::parse_connection_limit( const std::string& limit_str) {
+      std::istringstream in(limit_str);
+      fc_dlog( logger, "parsing connection endpoint with locale ${l}", ("l", std::locale("").name()));
+      in.imbue(std::locale(""));
+      double limit{0};
+      in >> limit;
+      size_t block_sync_rate_limit = 0;
+      if( limit > 0.0f ) {
+         std::string units;
+         in >> units;
+         std::regex units_regex{"([KMGT]?[i]?)B/s"};
+         std::smatch units_match;
+         std::regex_match(units, units_match, units_regex);
+         try {
+            block_sync_rate_limit = boost::numeric_cast<size_t>(limit * prefix_multipliers.at(units_match[1].str()));
```

Comment:
Test cases that fail this:
` --p2p-listen-endpoint 0.0.0.0:9876:8966Tbs` => rate limit of 8966 bytes
`--p2p-listen-endpoint 0.0.0.0:9876:8966Tb/s` => rate limit of 8966 bytes 
`--p2p-listen-endpoint 0.0.0.0:9876:123abc` => rate limit of 123 bytes.
We should generate an error and refuse to start if not a valid type specified.

`--p2p-listen-endpoint 0.0.0.0:9876:8966TB/s` => 
`error 2023-08-26T13:38:40.503 nodeos    net_plugin.cpp:4269           operator()           ] net_plugin::plugin_startup failed to listen on 0.0.0.0:9876:8966TB/s, Incorrect plugin configuration`
Even `1TB/s` and `1TiB/s` do not work.

Reply:
Now enforcing the full throttle format.

Added extra listen port to throttle test with 1TB/s limit to verify a non-empty prefix works and net_plugin successfully binds to the address.

Reply:
This still produces odd error messages:
`net_plugin::plugin_startup failed to listen on 0.0.0.0:9876:123gb/s, Incorrect plugin configuration`
It should report that `123gb/s` is not a valid rate limit option.

I think we need a unittest that tests all these various settings.

Reply:
This still needs fixed.

Reply:
This still reports the generic error:
`net_plugin::plugin_startup failed to listen on 0.0.0.0:9876:123gb/s, Incorrect plugin configuration` instead of something like `123gb/s` is not a valid rate limit.

Reply:
Fixed in https://github.com/AntelopeIO/leap/pull/1540/commits/6db4ad8aa535d0d4aef970ed4b74ef485be537d5.